src/main/java/org/javarosa/core/model/utils/DateUtils.java

```diff
@@ -440,13 +484,29 @@ private static boolean parseDateAndStore(String dateStr, DateFields df) {
     }
 
     public static Date parseTime(String str) {
+        if (timezoneOffset() != -1  && !str.contains("+") && !str.contains("-") && !str.contains("Z")) {
+            str = str + getOffsetInStandardFormat(timezoneOffset());
+        }
+
         DateFields fields = new DateFields();
         if (!parseTimeAndStore(str, fields)) {
             return null;
         }
         return getDate(fields);
     }
 
+    private static String getOffsetInStandardFormat(int offsetInMillis) {
+        int hours = offsetInMillis / 1000 / 60 / 60;
```

Comment:
How does this work for (ridiculous) timezones with `30` or `45` minute offsets form UTC?

Reply:
Ah wow good catch, will fix